components/support/sql/src/conn_ext.rs

```diff
@@ -213,3 +217,72 @@ impl<'conn> ConnExt for UncheckedTransaction<'conn> {
         &*self
     }
 }
+
+/// This transaction is suitable for when a transaction is being used only
+/// for performance when updating a larger number of rows (ie, when the
+/// transaction is not necessary for data integrity). You can specify a
```

Comment:
I think it's OK for integrity, too, as long as you don't call `maybe_commit` until you're actually ready to commit? As in, if you have a sequence of operations like:

```txt
WRITE A
WRITE B
MAYBE COMMIT
WRITE C
WRITE D
MAYBE COMMIT
```

We might commit (A, B) and (C, D) in two different transactions, or we might commit (A, B, C, D) in one. But you should only call `maybe_commit` after (A, B) if you're OK with either being interrupted after, or other readers seeing (A, B) without you having written (C, D). If that's not OK, you should wait to call `maybe_commit` until you've written (C, D), too...with the caveat that this might take longer than `commit_after`, and lose the advantage of using a `TransactionForPerf` over a vanilla `UncheckedTransaction`.

IOW, assume that `maybe_commit` will _always_ commit, and treat the fact that it doesn't, and might roll up multiple transactions in one, as an optimization...which is, indeed, why `TransactionForPerf` exists!

Reply:
Hm, that makes sense to me, and I hadn't thought about it like this.

Reply:
> I think it's OK for integrity, too

I need to think about rewording this, but the pedant in me thinks your example *is* just about perf. Eg, 
```
BEGIN IMMEDIATE
WRITE A
WRITE B
WRITE C
WRITE D
COMMIT
```

when you could COMMIT between B and C, but choose not to, you are probably making that choice based on performance.

I do understand your point though, so I'll try and make that clearer in the comments.